nullability/v0.1/nullability-v0.1.graphql

```diff
@@ -0,0 +1,61 @@
+"""
+Indicates that a field is only null if there is a matching error in the `errors` array.
+In all other cases, the field is non-null.
+
+Tools doing code generation may use this information to generate the field as non-null.
+
+This directive can be applied on field definitions:
+
+```graphql
+type User {
+    email: String @semanticNonNull
+}
+```
+
+It can also be applied on object type extensions for use in client applications that do
+not own the base schema:
+
+```graphql
+extend type User @semanticNonNull(field: "email")
+```
+
+Control over list items is done using the `level` argument:
+
+```graphql
+type User {
+    # friends is nullable but friends[0] is null only on errors
+    friends: [User] @semanticNonNull(level: 1)
+}
+```
+
+@param field the name of the field if applied to an object definition or null if applied
+to a field definition
+
+@param level in case of a list type, level is the dimension where to apply the modifier,
+starting at 0 if there is no list.
+If level is null, the modifier is applied to all levels
+"""
+directive @semanticNonNull(field: String = null, level: Int = null) repeatable on FIELD_DEFINITION | OBJECT
+
+"""
+Indicates that a field acts as an error boundary in case of a GraphQL error.
+
+By default, the first GraphQL error throws and fails the whole response.
+
+@param level in case of a list type, level is the dimension where to apply the modifier,
+starting at 0 if there is no list.
+If level is null, the modifier is applied to all levels
+"""
+directive @catch(to: CatchTo! = RESULT, level: Int = null) repeatable on FIELD
+
+enum CatchTo {
+    NULL,
+    RESULT,
```

Comment:
Naming: `RESULT` makes sense when you know that it means it will be generated as a `FieldResult<T>` in Kotlin. But wondering if that's clear otherwise. Maybe `RESULT_TYPE` or `WRAP`?

Reply:
Ideally we have language like this in the spec that we can educate about and reuse (see also [this Discord discussion](https://discord.com/channels/625400653321076807/862834364718514196/1179200983314726963))

As of now, agreed this can be confusing but I'm not sure `WRAP` is more clear and `RESULT_TYPE` is more verbose and I don't feel the extra verbosity adds much more. This is 100% a preference thing though so very open to changing this at this point.

What about adding a description to the enum value as a first step? If that's not enough, we'll change to `RESULT_TYPE`?

Reply:
Description added in https://github.com/apollographql/specs/pull/36/commits/6f3580ee62e20b17250f87f14ccf3c55b12fbe94